### PR TITLE
fix remix button

### LINKED ISSUE #10: add remix
remix paste
> take an existing paste content, change it to be your own
> and store it as its own paste

- on the page that reads an existing paste, add a "remix" button
- when clicked, the content becomes editable, text area with the value being the content
- "save" button is now visible

### what should the url become when we hit remix?
- viewing a paste: `https://pastebin.com/6j0z49TG`
- when we hit the remix button: `https://pastebin.com/remix/6j0z49TG`
- when you hit save: `https://pastebin.com/<new thing>`

### PR DESCRIPTION
**Pull request by anarellano**
followed the insturction and teseted it out

### CODE REVIEW

**Comment by anarellano** [comment]
when I deleted the branches the previous commits disappeared, so I  had to get back the code manually from closed pull requests.

it also created a lot of fileStorage files on accident, idk if you want to delete them or not?

**Review by evanugarte — COMMENTED** [review]
how about instead `/remix/:name` renders a new component `RemixPaste` instead

`RemixPaste` checks the id when the page loads and renders a textarea with the initial value of whatever the paste is

File: server/api.py [hunk 1]
```diff
@@ -71,6 +72,11 @@ def delete(name: str):
         print("Could not delete")
 
 
+@app.post("/edit/{name}")
```
Comment (evanugarte) [inline server/api.py]: i dont think we need this

File: server/sqlite_commands.py [hunk 2]
```diff
@@ -57,3 +57,15 @@ def fetch_all(db_path):
     print("hey", all_records, flush=True)
     db.close()
     return all_records
+
+
+def editPage(db_path, name):
```
Comment (evanugarte) [inline server/sqlite_commands.py]: also dont think we need this

File: server/api.py (hunk 1 above)
Comment (anarellano) [reply server/api.py]: resolved, thank you

File: server/sqlite_commands.py (hunk 2 above)
Comment (anarellano) [reply server/sqlite_commands.py]: resolved, thank you

File: client/src/components/remixPaste.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        },
+        body: JSON.stringify({ text: message }),
+      });
+
+      if (!res) {
+        throw new Error("No response saved");
+      }
+
+      const msg = await res.json();
+      console.log("Return Message:", msg);
```
Comment (evanugarte) [inline client/src/components/remixPaste.js]: probably dont need this

File: client/src/components/remixPaste.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      const res = await fetch("http://localhost:8000/store_paste", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ text: message }),
+      });
+
+      if (!res) {
+        throw new Error("No response saved");
```
Comment (evanugarte) [inline client/src/components/remixPaste.js]: why would we throw an error? maybe instead we render something to the user? like have a hook for an error message, and set it to true

check out the example in this code on how we render errors to a user

https://github.com/SCE-Development/Clark/blob/d6b9ce481002ad72b0e339c2fbba5ab81649745f/src/Pages/Speaker/Speaker.js#L69

**Review by evanugarte — COMMENTED** [review]
when we click "remix paste", the url should update to `/remix/<id of paste we are remixing`

when we save the remixed paste, the url should then update to `/view/<id of paste>`

we should not be using a hook inside the text of the remix/save button, let me know if this makes sense

File: client/src/components/remixPaste.js (hunk 3 above)
Comment (anarellano) [reply client/src/components/remixPaste.js]: I used this to change the name of the url, but if it was the log you were reffering to then it has been deleted

File: client/src/components/remixPaste.js (hunk 4 above)
Comment (anarellano) [reply client/src/components/remixPaste.js]: Chnaged the error, ty

Comment (anarellano) [reply client/src/components/remixPaste.js]: Took out the hooks for save and remix button, works fine still

File: client/src/pages/formatFile.js [hunk 5]
```diff
@@ -1,10 +1,15 @@
 import React, { useEffect, useState } from "react";
 import { useLocation, useParams } from "react-router-dom";
-import History from "../components/history.js";
 
 const formatFile = () => {
   const location = useLocation();
+
   const [data, setData] = useState(null);
+  const [title, setTitle] = useState("");
+  const [remix, setRemix] = useState("Remix");
```
Comment (evanugarte) [inline client/src/pages/formatFile.js]: this is unused now, make it hardcoded remix in the button

File: client/src/pages/formatFile.js [hunk 6]
```diff
@@ -27,12 +34,73 @@ const formatFile = () => {
...
+  // how about instead /remix/:name renders a new component RemixPaste instead
+  // RemixPaste checks the id when the page loads
+  //and renders a textarea with the initial value of whatever the paste is
+
+  // when the Remix Button is pressed
+  // Navigates to the new component remixPaste.js
+  // once it gets to the page, the text area is already open with the previous message inside
+  // when the buttone is pressed the page renders to the new page ../{name}
+
+  const handleClick = async (message, title) => {
```
Comment (evanugarte) [inline client/src/pages/formatFile.js]: we can remove this? this and the other comment about the unused hook also goes for remixPaste.js

File: server/api.py [hunk 7]
```diff
@@ -53,6 +53,7 @@ def find(name: str):
 @app.get("/history_component")
 def history_component():
     data = fetch_all(args.sqlite_file_path)
+    print(data)
```
Comment (evanugarte) [inline server/api.py]: remove

File: server/args.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--sqlite-file-path",
+        default="pastes.db",
+        help="The path to the SQLite database file.",
+    )
+
+    parser.add_argument("--host", default="localhost", help="add an IP adress")
```
Comment (evanugarte) [inline server/args.py]: look at the below for better help text for host and port

https://github.com/SCE-Development/cleezy/blob/a37893f33252857e085bfbe06376b137129bf992/modules/args.py#L12

File: server/args.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import argparse
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--sqlite-file-path",
+        default="pastes.db",
+        help="The path to the SQLite database file.",
```
Comment (evanugarte) [inline server/args.py]: say what the default value is in the help text

File: client/src/pages/formatFile.js (hunk 5 above)
Comment (anarellano) [reply client/src/pages/formatFile.js]: Resolved, ty

File: client/src/pages/formatFile.js (hunk 6 above)
Comment (anarellano) [reply client/src/pages/formatFile.js]: resolved, ty

File: server/api.py (hunk 7 above)
Comment (anarellano) [reply server/api.py]: resolved, ty

File: server/args.py (hunk 8 above)
Comment (anarellano) [reply server/args.py]: Changed the help text based on the link, ty

File: server/args.py (hunk 9 above)
Comment (anarellano) [reply server/args.py]: Changed it

File: server/args.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import argparse
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--sqlite-file-path",
+        default="pastes.db",
+        help=" The path to pastes.db",
```
Comment (evanugarte) [inline server/args.py]: how about "path to the sqlite file to store paste metadata. defaults to ./pastes.db"

File: client/src/components/remixPaste.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+          }
+          const file = await res.json();
+
+          setData(file);
+          setTitle(file.file[1]);
+          setMessage(file.show_file);
+        }
+      } catch (error) {
+        console.error(error);
+      }
```
Comment (evanugarte) [inline client/src/components/remixPaste.js]: instead dont throw an error, but keep the catch, and in there set data and title to null

File: client/src/pages/formatFile.js [hunk 12]
```diff
@@ -25,14 +37,63 @@ const formatFile = () => {
...
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ text: message }),
+      });
+
+      if (!res) {
+        throw new Error("No response saved");
+      }
```
Comment (evanugarte) [inline client/src/pages/formatFile.js:65]: also here

**Review by evanugarte — APPROVED** [review]

File: client/src/components/remixPaste.js (hunk 11 above)
Comment (anarellano) [reply client/src/components/remixPaste.js]:
```suggestion

          if (!res) {
            setData(null)
            setTitle(null)
                                }
          const file = await res.json();

          setData(file);
          setTitle(file.file[1]);
          setMessage(file.show_file);
        }
      } catch (error) {
        console.error(error);
      }
```

is this what you meant?

File: client/src/pages/formatFile.js (hunk 12 above)
Comment (anarellano) [reply client/src/pages/formatFile.js:65]: pushed the commits please review

**Review by evanugarte — APPROVED** [review]

**Pull request merged**